traitsui/editor_factory.py

```diff
@@ -184,9 +187,13 @@ def _get_toolkit_editor(cls, class_name):
                 factory_class = editor_factory_classes[index]
                 editor_file_name = os.path.basename(
                                 sys.modules[factory_class.__module__].__file__)
-                return toolkit_object(':'.join([editor_file_name.split('.')[0],
-                                             class_name]), True)
+                object_ref = ':'.join([editor_file_name.split('.')[0],
+                                       class_name])
+                return toolkit_object(object_ref, True)
             except Exception as e:
+                if TRAITS_DEBUG:
+                    msg = "Can't import toolkit_object '{}': {}"
+                    print(msg.format(object_ref, e))
```

Comment:
The point of turning on `TRAITS_DEBUG` is to reraise the exception. If you don't want to reraise the exception ever, just unconditionally use `logger.debug()`.